src/main/scala/io/dronekit/cloud/AWSWrapper.scala

```diff
@@ -156,27 +156,28 @@ class AWSWrapper(S3Client: AmazonS3Client = S3.client)
     * @param s3url The location in S3 of the object
     * @return A HTTP url for the object. Will time out!
     */
-  def signUrl(s3url: S3URL, expiry: Date = new DateTime(DateTimeZone.UTC).plusDays(7).toDate, filename: Option[String] = None): String = {
+  def signUrl(s3url: S3URL, expiry: ZonedDateTime = ZonedDateTime.now(ZoneId.of("UTC")).plusDays(7), filename: Option[String] = None): String = {
     filename match {
-      case None => S3Client.generatePresignedUrl(s3url.bucket, s3url.key, expiry).toString
+      case None => S3Client.generatePresignedUrl(s3url.bucket, s3url.key, Date.from(expiry.toInstant)).toString
       case Some(name) => {
         val request: GeneratePresignedUrlRequest = new GeneratePresignedUrlRequest(s3url.bucket, s3url.key, HttpMethod.GET)
         val headerOverrides: ResponseHeaderOverrides = new ResponseHeaderOverrides()
         headerOverrides.setContentDisposition(s"attachment; filename=$name")
-        request.withResponseHeaders(headerOverrides).withExpiration(expiry)
+        request.withResponseHeaders(headerOverrides).withExpiration(Date.from(expiry.toInstant))
         S3Client.generatePresignedUrl(request).toString
       }
     }
   }
 
+  def streamUpload(s3url: S3URL): Sink[ByteString, Future[CompleteMultipartUploadResult]] = {
+    multipartUploadTransform(s3url).to(Sink.head)
```

Comment:
You are correct, I needed to consume all of the UploadPartRequests for the upload to complete if there was more than one part.